lectures/Lecture3.3_ConvNets.ipynb

```diff
@@ -0,0 +1,2382 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Once installed (or, once opened this notebook in Colab), you can start using PyTorch with a simple import statement:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import torch"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## **Tensors, Datasets, & DataLoaders** <a class=\"anchor\" id=\"tensors-datasets--dataloaders\"></a>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In the following, we will use PyTorch to build a CNN for image recognition. Therefore, we will need some images to work with. \n",
+    "\n",
+    "PyTorch has a companion library that contains classes and functions to work with images. This companion library is called `torchvision`.\n",
+    "\n",
+    "Among other things, `torchvision` contains a `datasets` package (a package is a collection of `.py` files), which provides tools to download and use publicly available image datasets. \n",
+    "\n",
+    "We will use the [MNIST dataset of handwritten digits](https://en.wikipedia.org/wiki/MNIST_database), which contains $70000$ handwriting samples of the numbers from $0$ to $9$, encoded as $28$ x $28$ grids of grayscale luminosity values:\n",
+    "\n",
+    "<p align=\"center\">\n",
+    "  <img src=\"./files/mnist-samples.png\" />\n",
+    "</p>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The code below downloads MNIST from its online repository. We do two separate downloads for the training and test datasets, assigning each to a separate variable.  "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import torchvision\n",
+    "import torchvision.transforms as transforms\n",
+    "\n",
+    "training_dataset = torchvision.datasets.MNIST(root=\"classifier data\", \n",
+    "                                              train=True, \n",
+    "                                              download=True, \n",
+    "                                              transform=transforms.ToTensor())\n",
+    "test_dataset = torchvision.datasets.MNIST(root=\"classifier data\", \n",
+    "                                          train=False, \n",
+    "                                          download=True, \n",
+    "                                          transform=transforms.ToTensor())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The variables that we created are _dataset objects_. A dataset object implements two crucially important methods:\n",
+    "\n",
+    "- `__len__()`, which allows users to get the dataset's length\n",
```

Comment:
We did not cover those special methods in the objects class, so I would keep their explicit mentioning out as it's distracting - sufficient to say you can index them and retrieve the Len with `len()`

Reply:
Ok